File: problem1.cpp
// Time Complexity :  O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : yes
// Any problem you faced while coding this :


// Your code here along with comments explaining your approach

// we push numbers in to stack. if there is an operation character,
// we pop elements from stack and perform operation and push again.
// at last stack.top() will be the answer.


class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        int n = tokens.size();
        stack<int>st;
        int ans = 0;
        for(auto& s:tokens)
        {
            if(s=="/" || s=="*" || s=="+" || s=="-")
            {
                int b = st.top();st.pop();
                int a = st.top();st.pop();
                if(s=="+") ans = a+b;
                else if(s == "-") ans = a-b;
                else if(s=="*") ans = a*b;
                else ans = a/b;
                st.push(ans);
            }
            else
            {
                int num = stoi(s);
                st.push(num);
            }
        }
        return st.top();
    }
};

File: Problem1.cs
// Time Complexity : O(n)
        // Space Complexity : O(n)
        // Did this code successfully run on Leetcode : Yes
        // Any problem you faced while coding this : No 
        public int EvalRPN(string[] tokens)
        {
            if (tokens == null || tokens.Length == 0) return 0;
            Stack<int> st = new Stack<int>();
            int result = 0;
            for (int i = 0; i < tokens.Length; i++)
            {
                int num;
                if (int.TryParse(tokens[i], out num))
                {
                    st.Push(num);
                }
                else
                {
                    int b = st.Pop();
                    int a = st.Pop();
                    switch (tokens[i])
                    {
                        case "+":
                            result = a + b;
                            break;

                        case "-":
                            result = a - b;
                            break;

                        case "*":
                            result = a * b;
                            break;

                        case "/":
                            result = a / b;
                            break;

                        default:
                            break;
                    }
                    st.Push(Convert.ToInt32(result));
                }
            }
            return st.Pop();
        }
